addon-resizer/nanny/nanny_lib.go

```diff
@@ -100,15 +118,24 @@ func PollAPIServer(k8s KubernetesClient, est ResourceEstimator, contName string,
 		expResources := est.scaleWithNodes(num)
 
 		// If there's a difference, go ahead and set the new values.
-		if !shouldOverwriteResources(int64(threshold), resources.Limits, resources.Requests, expResources.Limits, expResources.Requests) {
+		overwrite, op := shouldOverwriteResources(int64(threshold), resources.Limits, resources.Requests, expResources.Limits, expResources.Requests)
+		if !overwrite {
 			log.V(4).Infof("Resources are within the expected limits. Actual: %+v Expected: %+v", *resources, *expResources)
 			continue
 		}
 
+		now := time.Now()
+		if (op == scaleDown && now.Before(lastChange.Add(scaleDownDelay))) ||
```

Comment:
Can you point me to the specific line in test where this thing is properly unit-tested. I don't see a line where we check what happens if last change is slightly before or slightly after scale****Delay. But maybe I'm missing something.

Reply:
I have extracted logic for checking resources for deployment and updating them to a separate method and added unit tests for this.
When it comes E2E tests I have build an image with addon-resizer, deployed it to a Kubernetes cluster first without any delay being set and then with scale up and scale down being set and I have scaled cluster from 3 to 10 nodes and then back to 3 nodes.